feat(authentication): Add Sign In / Sign Up Page and Cookie Based Authentication with User Friendly design

seraGeorge opened the pull request:
This pull request introduces significant enhancements to the authentication and user experience features. The key updates include:

**1. Sign Up and Sign In Pages:**

Implemented dedicated pages for user registration (Sign Up) and authentication (Sign In).
Added the functionality to log out from any page, providing users with a seamless and convenient experience.

**2. Cookie-Based Authentication:**

Implemented robust cookie-based authentication for secure and efficient user sessions.
Enhances the overall security and usability of the authentication process.

**3. UI Upgrade for Improved User Experience:**

Revamped the user interface to ensure a visually appealing and user-friendly design.
These UI enhancements contribute to an overall improved user experience, making navigation more intuitive.

These changes collectively aim to elevate the authentication process and enhance the user interface, ensuring a smoother and more secure experience for our users.

seraGeorge committed:
[add]:type change in actions

seraGeorge committed:
[add]: responsive design for emplyee card list

seraGeorge committed:
[add]: search bar and select button to the card layout

seraGeorge committed:
[add]: pagination for employee listing

seraGeorge committed:
[add]: pagination from the select component in the employee table header

seraGeorge committed:
[add]: table sorting

seraGeorge committed:
[fix]: pagination select component change if page in the pagination bubble change
[add]: firstname as parameter for sorting

seraGeorge committed:
[change]: tooltip hovering action simplified

seraGeorge committed:
[fix]: pagination depth reached issue

[move]: pagination into employeetable

seraGeorge committed:
[remove]:  console in details component
[change]: params to be nullable

Commit:
[Add]: Added progress bar in form

seraGeorge committed:
[add]: basic infinite loading

seraGeorge committed:
merged from main

seraGeorge committed:
[add]: for debugging

seraGeorge committed:
[rename]: employees in store to employeesForList

ashwin8448 committed:
[edit]: Form page

ashwin8448 committed:
[edit]: Removed comments

seraGeorge committed:
[fix]: infinite page loading

the grid button is clicked, the page loads as initial loading

Commit:
edit:[form pag]

Commit:
[edit]: Photo upload

seramg committed:
[change]: restructured the data structure

seramg committed:
[move]:employee data fetching in the parent component

seramg committed:
[add]: skills filtering function

seramg committed:
[remove]: unwanted skills filtering setting on mount

seramg committed:
[move]: search and select all in the employee listing
[redirect]: to page 1 when filtering is done
[add]: table overflow with border

seramg committed:
[add]:searching

ashwin8448 committed:
[edit]: Form page

seramg committed:
[fix]: filtering in employee card list

seramg committed:
[remove]: timeout delay

seraGeorge committed:
[add]:margin for infinite-scroll-loader
[remove]: unwanted print
[remove]: infinite loading state and passed loading from search params
[add]: type in search params

seraGeorge committed:
Merge branch 'employee-listing' of https://github.com/ashwin8448/HRM_Redux into employee-listing

seraGeorge committed:
[fix]: infinite loading with filtering and searching

seraGeorge committed:
Merge branch 'employee-listing' of https://github.com/ashwin8448/HRM_Redux into employee-listing

seraGeorge committed:
[add]: employee card list

seraGeorge committed:
[remove]: skill loader
and adjusted card list

seraGeorge committed:
[fix]:started fixing employee card list select all

Commit:
[add]: form config

Commit:
[edit]: form validation

seraGeorge committed:
[add]: necessary log for debugging

seraGeorge committed:
Merge branch 'employee-listing' of https://github.com/ashwin8448/HRM_Redux into employee-listing

seraGeorge committed:
[add]: logic for checkbox deleting

seraGeorge committed:
[add]: logic for confirm delete action

seraGeorge committed:
[add]: minor fix to the deete checkbox

seramg committed:
optimised the code

seramg committed:
[remove]: filter select list

ashwin8448 committed:
[Edit]: Photo upload.

seraGeorge committed:
Optimised the code

seramg committed:
[add]: tooltip height to 100px
[remove]:  table loader top to 50%

seramg committed:
[add]: access token to cookies

seramg committed:
[add]: search params updation on deleting modal

seramg committed:
[remove]: unwanted consoles

ashwin8448 committed:
[Edit: User image upload]

Commit:
[edit]:photo upload

seraGeorge committed:
[add]: design changes to tooltip

seraGeorge committed:
Merge branch 'employee-listing' of https://github.com/ashwin8448/HRM_Redux into employee-listing

seraGeorge committed:
[rename]: IEmployee to IReceivingEmployee

seraGeorge committed:
Merge branch 'employee-listing' of https://github.com/ashwin8448/HRM_Redux into auth-cookie

ashwin8448 committed:
[edit]: Interface, form page and form submission

seraGeorge committed:
[add]:login logic
now when access token is deleted, navigated to login page
but when token is present it is navigated to home page

Commit:
[edit]: Added new interfaces to manage data

Commit:
[edit]:Removed unused components

Commit:
[fix]: Skillchip bug

Commit:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into formPage

seraGeorge committed:
[add]: display parameter on search url and
[remove]: unwanted parameters in url

seraGeorge committed:
[add]: sort button

Commit:
[edit]: API Interceptor

Commit:
[edit]: View Employee page

Commit:
[edit]: Deleted unused filses

Commit:
[delete]: Unused files

ashwin8448 committed:
Update deploy.yml

seraGeorge committed:
[change]:  branch name in deploy.yml

seraGeorge committed:
[remove]: unwanted css

seraGeorge committed:
[remove]: unwanted todo
[change]: repo name in routes and vite config
[remove]: empty file.

seramg committed:
Merge branch 'employee-listing' of github.com:ashwin8448/HRM_Redux into auth-cookie

seramg committed:
[add]: logic for automatic logout

seraGeorge committed:
[change]: tableactions to actions

seraGeorge committed:
[add]: UI Improvements for employee listing page

seraGeorge committed:
[add]: null check for filterselect
[add]: global-width for header
[remove]: sequential loading for dropdown
[rename]: actions->listingactions; actionsbar->filteractions
[add]: handle outside click. if clicked outside dropdown goes off

seraGeorge committed:
[add]: padding to sort order

seraGeorge committed:
[add]: names to the delete modal

seraGeorge committed:
Merge branch 'employee-listing' of https://github.com/ashwin8448/HRM_Redux into auth-cookie

seraGeorge committed:
[add]: protected route for the login

seramg committed:
[remove]: token in store

seramg committed:
[fix]:infinite loading on reloading

seramg committed:
[fix]: infinite loading on reloading

seramg committed:
[remove]: unnecessary code

seramg committed:
[add]:login authentication using refresh token

seramg committed:
[add]:onclick to view page and increase checkbox size

ashwin8448 committed:
[edit]: Changes made as per review

seramg committed:
[remove]: handleActivelisting

seraGeorge committed:
Merge pull request #1 from ashwin8448/employee-listing

feat(employee listing): Add Employee Listing Page with Dual Views and Advanced Functionalities for Enhanced User Experience

seramg committed:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into formPage

seramg committed:
[add]: changes to redux

seraGeorge committed:
[add]: color change for disabled btn
[add]: key for employee name list item
[add]: status range to accomodate 201 status
[add]: skillsdata,deptdata,rolesdata for actions
[add]: setloading type action
[remove]: store.dispatch and used dispatch
[remove]: aliasing for action.payload
[add]: action type in reducers
[add]: useAppSelector instead of useSelector
[remove]: unnecessary props
[add]:link to the edit employee
[add]: navigation after adding employee
[add]: const instead of let in for loop
[remove]: border in view employee
[remove]: unnecessary div in view employee layout

seraGeorge committed:
[remove]: errors for deployment
unnecessary imports
[add]: deletemodal in employeeviewlayout as comment

seraGeorge committed:
[add]: develop branch in deploy.yml

seraGeorge committed:
Merge branch 'develop' of https://github.com/ashwin8448/HRM_Redux into auth-cookie

seraGeorge committed:
[remove]: unnecessary imports and unnecessary texts
[remove]: fetchdropdowndata in layout
[add]: helper functions for getting cookies

Commit:
[edit]: Made changes as per review

Commit:
[edit]: Dropdown data fetch removed from layout

Commit:
[add]: Delete functionality added to view employee page.

seraGeorge committed:
[add]: snackbar on selection and added to the employee listing page
[add]: dummy img for employees with no img
[move]: checkbox to helper
[add]: margin to delete modal elements
[add]: index as key in employeenameslist
[add]: ui improvements for listing actions
[add]: z index for sort dropdown
[add]: label as tooltip msg
[add]: center loader for view and form page
[add]: handlecheckboxlist in helper fn

seraGeorge committed:
Merge branch 'employee-listing' of https://github.com/ashwin8448/HRM_Redux into formPage

seraGeorge committed:
[add]: logo to the header
[add]: ui changes to the employee view page
[modularise]: the view employee page

Commit:
[FIX]: Navigation issue and refresh token issue

Commit:
[FIX]: Navigation issue and refresh token issue

Commit:
Merge branch 'formPage' of github.com:ashwin8448/HRM_Redux into formPage

Commit:
[Add]: User login page

seraGeorge committed:
[change]: email to name

seraGeorge committed:
Merge branch 'formPage' of https://github.com/ashwin8448/HRM_Redux into formPage

seraGeorge committed:
[add]: re enter password check.
[improve]:  Ui

seraGeorge committed:
move actiontypes.
add onchange event type

seraGeorge committed:
[add]: null value to employee id

Commit:
[fix]: Add employee bug

seraGeorge committed:
Merge branch 'main' of https://github.com/ashwin8448/HRM_Redux into develop

seraGeorge committed:
[add]: favicon to the assets folder

seraGeorge committed:
[add]: correct path tothe favicon

seraGeorge committed:
[fix]: login,filter actions ui issue
[add]: correct path to the assets

seraGeorge committed:
[add]:100vh issue on mobile

seraGeorge committed:
[add]: responsive

seraGeorge committed:
[remove]: unnecessary path navigation
[change]: condition for active listing

seraGeorge committed:
[add]: ui modifications

lekshmi615213 commented on src/components/DeleteModal/DeleteModal.tsx:
```diff
@@ -0,0 +1,107 @@
+  changeDltModalOpenStatus: () => void;
+  idArrayToDlt: string[];
+}) {
+  const [searchParams, setSearchParams] = useSearchParams({ page: "1" });
```
can set a constants for default per page/ page number

seraGeorge replied:
Resolved in 6b30411148cf8a8a5256dd6aa54c2bbe69d19085

lekshmi615213 commented on src/components/DeleteModal/DeleteModal.tsx:
```diff
@@ -0,0 +1,107 @@
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const confirmDlt = async () => {
```
good to use meaningful methodName/variableName, add proper types for arguments/ return types etc

seraGeorge replied:
Resolved in 8ec226b3b8c25dd4bfda0abba9694fa60decda75

lekshmi615213 commented on src/components/DeleteModal/DeleteModal.tsx:
```diff
@@ -0,0 +1,107 @@
+      setSearchParams({
+        ...Object.fromEntries(searchParams.entries()),
+        ...pageValue,
+      });
```
can we move this code to other util file for this

seraGeorge replied:
Resolved in 4167e78190a3c40f9348ec2ce3fed28de1614605

lekshmi615213 commented on src/components/DeleteModal/DeleteModal.tsx:
```diff
@@ -0,0 +1,107 @@
+        className="close-btn"
+        onClick={changeDltModalOpenStatus}
+      ></Button>
+      <h2 className="delete-modal-heading">{DELETE_MODAL_HEADING}</h2>
```
hope we have a H1 as we have to keep hierarchy for h tags, also better to use semantic tags than multiple DIVs

seraGeorge replied:
used h1 ffd3b6190fb2d2b9566822483567e76a22f5df06
changed div to span 90caa6d435413605195685acf34d14fa64247ce5

lekshmi615213 commented on src/components/DeleteModal/deleteModal.ts:
```diff
@@ -0,0 +1,104 @@
+    color: ${colors.WHITE_COLOR};
+
+    &:hover {
+      background-color: rgba(200, 30, 30, 1) !important;
```
why important?

seraGeorge replied:
Resolved in 76588e532b342cc9b73d8157c4f33e905b36a284

lekshmi615213 commented on src/components/DeleteModal/deleteModal.ts:
```diff
@@ -0,0 +1,104 @@
+    justify-content: flex-start;
+  }
+  .delete-btn {
+    background-color: #e02424 !important;
```
color variable?

seraGeorge replied:
Resolved in 3a31f6702b59f31ff9282876fe39e4adcddd7b2b

lekshmi615213 commented on src/components/Details/details.ts:
```diff
@@ -0,0 +1,44 @@
+    align-items: center;
+  }
+
+  .skills-tooltip{
```
indentation

seraGeorge replied:
Resolved

lekshmi615213 commented on src/components/FilterSelect/FilterSelect.tsx:
```diff
@@ -0,0 +1,40 @@
+          isSearchable={true}
+          options={options}
+          placeholder={
+            <div className="placeholder overflow-ellipsis">{placeholder}</div>
```
div needed?

seraGeorge replied:
Resolved in eb1b2b519d617b6f50214694a2b45a92b3d1aa4d

lekshmi615213 commented on src/components/FilterSelect/FilterSelect.tsx:
```diff
@@ -0,0 +1,40 @@
+          }
+          isMulti={isMulti}
+          styles={selectStyleComponent()}
+          onChange={(selectedOption) => {
```
: ISelectOptionProps[] ?

lekshmi615213 commented on src/components/Footer/footer.ts:
```diff
@@ -0,0 +1,15 @@
+    width: 100%;
+    background-color:  ${colors.WHITE_COLOR};    
+    border-top: 1px solid ${colors.LIGHT_GRAY_COLOR};
+    padding: 20px 0 20px 0;
```
padding-top:20px; padding-bottom:20px ?

seraGeorge replied:
Resolved in 642a3efafd4530938de95580638849f0b1cbf8dc

seraGeorge committed:
[add]: photo container design  placed commonly

seraGeorge committed:
[move]: font sizes and weight to new file

seraGeorge committed:
[move]: search params to a  new file

seraGeorge committed:
[add]: text styles and styled components

Commit:
[edit]: Deployment error fix

seraGeorge committed:
[add]: vercel.json for deployment in vercel

seraGeorge committed:
[rectify]: image not found

seraGeorge committed:
[rectify]: href location

seraGeorge committed:
[rectify]: name of repo

seramg committed:
[remove]: toastcontainer from app.tsx
[change]: folder structure
[change]: active chip to active dot
[add]: checkbox styled component
[add]: adjustments to ui and rectified ui mistakes
[remove]: empty files
[improved]: work experience computation
[remove]: saving location state in routing

seramg committed:
[add]: edit button to card

seramg committed:
[change]: typo in filter button
[add]: responsiveness in employee view page

seramg committed:
[fix]: deployment

seramg committed:
[add]: minor change

seramg committed:
[add]: Minor change in route base.

seramg committed:
[add]:Minor change in vite config.

seraGeorge committed:
[add]: routing to the correct location entered after getting logged out

seraGeorge committed:
[add]: multi colors to skills chip list and add ui improvements

ashwin8448 committed:
[FIX]: Form page UI issue

seraGeorge committed:
[remove]: styled link
[add]: page change on url change
[add]: Ui improvements

ashwin8448 committed:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into formPage

ashwin8448 committed:
[Add]: media.ts file

seraGeorge committed:
[add]: MUI tooltip component
[rename]: chip styled component for active chip
[add]: hover styles for checkbox and active dot

seraGeorge committed:
[add]: 1st child of table row has white background colour

seraGeorge committed:
[add]: table row background color on hover

seraGeorge committed:
[add]: hover effect color on table rows

seraGeorge committed:
[add]: delete button to the table row

seraGeorge committed:
[add]: export button to more actions

seraGeorge committed:
[add]: dev dependency of react-csv

Commit:
Initial Commit

Commit:
[edit]: Implementing role based access

lekshmi615213 commented on src/components/ActiveDot/activeDot.ts:
```diff
@@ -0,0 +1,20 @@
+  width: 10px;
+  transform:  scale(1);
+  transition: transform 0.2s;
+  z-index: 0;
```
why this 0 value needed

seraGeorge replied:
Resolved in b4551ebb492936dce5682ce1c6defe8feaa5b196

lekshmi615213 commented on src/components/Checkbox/Checkbox.tsx:
```diff
@@ -0,0 +1,46 @@
+      }
+      checked={
+        employeeId
+          ? deleteCheckBoxesList.checkedBoxesList.includes(employeeId)
```
these much conditions needed? if so , please get the state to a const and use it here

seraGeorge replied:
Resolved in 70b4934300f70439e5aca8f8c21bc6affc0a87e5

lekshmi615213 commented on src/components/DeleteModal/DeleteModal.tsx:
```diff
@@ -0,0 +1,116 @@
+  idArrayToDlt,
+}: {
+  changeDltModalOpenStatus: () => void;
+  idArrayToDlt: string[];
```
idArrayToDelete? try to use proper names, check other places too

seraGeorge replied:
Resolved in 8ec226b3b8c25dd4bfda0abba9694fa60decda75

lekshmi615213 commented on src/components/DeleteModal/DeleteModal.tsx:
```diff
@@ -0,0 +1,116 @@
+        })
+      );
+
+      if (getUrlType(location.pathname) === "view-employee") navigate("/");
```
I think this is not a good idea to relay on url path name, can we use some other way

lekshmi615213 commented on src/components/DeleteModal/DeleteModal.tsx:
```diff
@@ -0,0 +1,116 @@
+      toast.error("Error deleting users", { toastId: "delete-user" });
+    } finally {
+      setConfirmDeleteLoader(false);
+      const isdisplayList = searchParams.get("display") === "List";
```
comment as above, also use defaultPageNumber constant

seraGeorge replied:
Resolved

lekshmi615213 commented on src/components/DeleteModal/DeleteModal.tsx:
```diff
@@ -0,0 +1,116 @@
+      </LabelStyles>
+      <ul className="employees-name-list">
+        {employeesNameList.map((employeesName, index) => (
+          <li key={index}>{employeesName}</li>
```
good to use id, rather than  array index as it might be constly

lekshmi615213 commented on src/components/DeleteModal/DeleteModal.tsx:
```diff
@@ -0,0 +1,116 @@
+          <li key={index}>{employeesName}</li>
+        ))}
+      </ul>
+      <div className="warning-container">
```
reduce the usage of Div

lekshmi615213 commented on src/components/Footer/Footer.tsx:
```diff
@@ -0,0 +1,19 @@
+
+function Footer() {
+  //Function to get the current Year
+  const getYear = () => {
```
always try to add proper types ([params and return)

lekshmi615213 commented on src/components/Input/Input.tsx:
```diff
@@ -0,0 +1,100 @@
+                        message: "This field is required",
+                      },
+                    })}
+                    max={config.validation?.max?.value} // for date input
```
can we add min too

lekshmi615213 commented on src/components/Input/Input.tsx:
```diff
@@ -0,0 +1,100 @@
+          case "date":
+          case "tel":
+          case "email":
+          case "textarea":
```
text are , we have <TextArea> can we use ?

lekshmi615213 commented on src/components/Input/PhotoInput.tsx:
```diff
@@ -0,0 +1,87 @@
+import PhotoInputWrapper from "./photoInput.ts";
+import { LabelStyles } from "../../core/constants/components/text/textStyledComponents.ts";
+
+const PhotoInput = ({ config }: { config: IInputProps }) => {
```
ImageUpload?

lekshmi615213 commented on src/components/ProgressBar/ProgressBar.tsx:
```diff
@@ -0,0 +1,41 @@
+        {steps.map((step, index) => (
+          <div key={index} className="step">
+            <LabelStyles
+              className={`step-number ${
```
u can check classnames package

lekshmi615213 commented on src/pages/EmployeeListing/Grid/EmployeeCardList/EmployeeCardList.tsx:
```diff
@@ -0,0 +1,129 @@
+  }, []);
+
+  useEffect(() => {
+    setPage(0);
```
can we try combing this with above line 86

lekshmi615213 commented on src/pages/EmployeeListing/List/components/Pagination/hook/usePagination.ts:
```diff
@@ -0,0 +1,88 @@
+    const paginationRange = useMemo(() => {
+
+        // Pages count is determined as siblingCount + firstPage + lastPage + currentPage + 2*DOTS
+        const totalPageNumbers = siblingCount + 5; // always 6
```
move these magic numbers to constants

seraGeorge replied:
Resolved in 70b4934300f70439e5aca8f8c21bc6affc0a87e5

lekshmi615213 commented on src/pages/EmployeeListing/components/ListingActions/MoreActions/MoreActions.tsx:
```diff
@@ -0,0 +1,149 @@
+    return () => {
+      document.body.style.overflow = "auto";
+    };
+  }, [deleteModal]);
```
can we try withour useeffect

lekshmi615213 commented on src/pages/EmployeeUpdate/Form.tsx:
```diff
@@ -0,0 +1,287 @@
+    if (employeeData)
+      for (const employeeProperty in employeeData) {
+        if (
+          (employeeProperty === "department" || employeeProperty === "role") &&
```
use some enums rather than checking with string

lekshmi615213 commented on src/pages/EmployeeUpdate/Form.tsx:
```diff
@@ -0,0 +1,287 @@
+        const newEmployee = await convertFormDataToIPostEmployees(formValues);
+        await postData(apiURL.employee, newEmployee);
+        // Display toast for success state
+        toast.success(`Added user ${newEmployee.firstName}`, {
```
what happens for failure cases?

lekshmi615213 commented on src/pages/EmployeeUpdate/Form.tsx:
```diff
@@ -0,0 +1,287 @@
+                    switch (activeSection) {
+                      case 1:
+                        validationStatus = await methods.trigger([
+                          "firstName",
```
use some dictionaries for props

seraGeorge committed:
[add]: set a constants for default page, defaultSearchParams and display value

seraGeorge committed:
[add]: use meaningful method names/variable names

seraGeorge committed:
[add]: h1 to the logo and error title

seraGeorge committed:
[add]: span instead of div

seraGeorge committed:
[add]: background color on hovering

seraGeorge committed:
[remove]: unnecessary important

seraGeorge committed:
[add]: color variables

seraGeorge committed:
[remove]: div for placeholder

seraGeorge committed:
[add]:  short hand for css and used less prop transfer for filter select
[rectify]: clear filters

seraGeorge committed:
[add]: common background color on hover

seraGeorge committed:
[remove]: unnecessary z-index prop

seraGeorge committed:
[add]: improved logic for checkstatus of checkbox

seraGeorge committed:
[move]: magic numbers to constants

ashwin8448 committed:
[edit]: Implementing role based access

Commit:
[edit]: Implementing role based access

ashwin8448 committed:
[Merge]: Merged latest changes from develop branch

ashwin8448 committed:
[edit]: Resolved merge conflicts

ashwin8448 committed:
[edit]: Working on access control

seramg committed:
[change]: records per page

seramg committed:
[fix]: no hover background color effect on loading

seramg committed:
[fix]:  padding in sort dropdown

seramg committed:
[add]: button hover effect

ashwin8448 committed:
[edit]: Validation error fix.

seramg committed:
[add]: login fix.
Page don't navigate to login page on reloading.

ashwin8448 committed:
[FIX]: Navigation bug

ashwin8448 committed:
[fix]: View employee page bug

seramg committed:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into access-control

seramg committed:
[optimise]: table css

seramg committed:
Merge branch 'access-control' of github.com:ashwin8448/HRM_Redux into access-control

seramg committed:
[fix]: add design changes

Commit:
[fix]: Form bugs and UI issues

Commit:
[fix]: UI issues

Commit:
[fix]: redux store bug fix

seraGeorge committed:
[fix]: infinite loading issue and search params

seraGeorge committed:
Merge branch 'access-control' of https://github.com/ashwin8448/HRM_Redux_Auth into access-control

seraGeorge committed:
[add]: additional condition for loading data

seramg committed:
[add]: infinite scroll fix

seramg committed:
[add]: select all fix

seramg committed:
[add]: design changes

ashwin8448 committed:
[Fix]: Form bug fix

ashwin8448 committed:
Merge branch 'access-control' of github.com:ashwin8448/HRM_Redux into access-control

seramg committed:
[add]: design changes for form

Commit:
[fix]: Login form issues

Commit:
[edit]: Removed console.log

seraGeorge committed:
[add]: back button color, box shadow to the card.
[improve]: pagination only after loading
[remove]: id from all possible pages

seraGeorge committed:
Merge branch 'develop' of https://github.com/ashwin8448/HRM_Redux_Auth into develop

Commit:
[fix]: doj validation bug

Commit:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into develop

seraGeorge committed:
[add]: meta tags to each page visited
[add]: padding to employee data
[add]: button design. a notification bubble will appear if skills ids are added.

seraGeorge committed:
Merge branch 'develop' of https://github.com/ashwin8448/HRM_Redux_Auth into develop

ashwin8448 committed:
[fix]: Edit form bug fixed

ashwin8448 committed:
[fix]: Filter UI Issues

seraGeorge committed:
[add]: tabIndex to forms

seraGeorge committed:
[add]: correct width to the last column

ashwin8448 committed:
[FIX]: Filter apply issue

ashwin8448 committed:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into develop

seramg committed:
[add]: tabIndx for login page

seramg committed:
[add]: filters nav bar doesn't show the filters if not clicked on apply

seramg committed:
[add]: tooltip for action buttons
 [add]: validation for email

seramg committed:
[fix]: no data design issue

seramg committed:
[add]: tootlip to profile btn

seramg committed:
[add]: background color of btn

seramg committed:
[add]: new logic for employees grid setup

seramg committed:
Revert "[add]: background color of btn"

This reverts commit fe7fce579b8f6b52074d92bbdc2b5db37a325691.

seramg committed:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into develop

seramg committed:
[revert]: change to setting employees

seramg committed:
[add]: fetch on delete

seramg committed:
[add]: debouncing in search

seramg committed:
[add]: debounceTimeout to the constants

ashwin8448 committed:
[Add]: Added avatars in form

ashwin8448 committed:
[edit]: Implemented login using email

Commit:
[add]: Added encrypted employee ID in URL.

seramg committed:
[add]: pagination works on clicking the bubble

seramg committed:
[add]; color to dummy  image

seramg committed:
[add]:  snackbar closes when clicked on no cancel or close btn in delete modal. and will be still open on clicking overlay

seramg committed:
[add]: disable scroll behavior on modal open. dropdown disappear on click away

ashwin8448 committed:
[edit]: Added progressive loader

ashwin8448 committed:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into develop

ashwin8448 committed:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into develop

seraGeorge committed:
[replace]: loader image with loader

seraGeorge committed:
[add]: progressive loader for images
and eliminated padding for icon buttons

seraGeorge committed:
[add]: very important star on input fields

seraGeorge committed:
[add]: zindex for modals

seraGeorge committed:
[add]: styling for login

seraGeorge committed:
[add]: small width for table name column

seraGeorge committed:
[add]: new loader

Commit:
[FIX]: Duplicate email issue

Commit:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into develop

seraGeorge committed:
[change]: width and height of steps and link in edit and add form

seraGeorge committed:
Merge branch 'develop' of https://github.com/ashwin8448/HRM_Redux_Auth into develop

Commit:
[FIX]: Duplicate email in add form

Commit:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into develop

seraGeorge committed:
[fix]: login submission error

seraGeorge committed:
Merge branch 'develop' of https://github.com/ashwin8448/HRM_Redux_Auth into develop

seraGeorge committed:
[change]: side filter ui

seraGeorge committed:
[change]: filter select don't have a clearable btn

seramg committed:
[add]: custom hook for overflow of text
